addons/point_of_sale/static/src/app/services/pos_store.js

```diff
@@ -2134,7 +2134,7 @@ export function register_payment_method(use_payment_terminal, ImplementedPayment
 export const posService = {
     dependencies: PosStore.serviceDependencies,
     async start(env, deps) {
-        return new PosStore({ traps: {}, env, deps }).ready;
+        return reactive(new PosStore({ traps: {}, env, deps }).ready);
```

Comment:
If internal changes need to trigger renders, the constructor should return the reactive version and everything in the internal code should write on the reactive.